### PR TITLE
Allow to compile collective for PT > 2.3

### PR DESCRIPTION
**Pull request by nelyahu**

### CODE REVIEW

File: deepspeed/comm/torch.py [hunk 1]
```diff
@@ -19,6 +19,10 @@
 DS_COMM_ALL_REDUCE_OFF = False
 DS_COMM_REDUCE_OFF = False
 
+def disable_compiler_collective(func):
```
Comment (loadams) [inline deepspeed/comm/torch.py:22]: FYI @nelyahu on formatting errors

**Review by tohtana — APPROVED** [review]

**Review by loadams — APPROVED** [review]

**Comment by loadams** [comment]
Hi @nelyahu - the nv-torch-latest-v100 we will fix, that's a known OOM.  The cpu-torch-latest I'm not sure why that's failing but looks unrelated to your changes.

**Comment by loadams** [comment]
Hi @nelyahu - I'm not sure if the changes are impacting the cpu-torch-latest since that's on torch > 2.3?  But it seems to always fail on this PR.  I haven't had any time to debug this just yet though

**Comment by nelyahu** [comment]
Seems like it is related to the PR it self. we will take a look into this locally and will update here

**Comment by loadams** [comment]
Closing in favor of #6899

**Pull request closed without merging**